satpy/modifiers/parallax.py

```diff
@@ -0,0 +1,408 @@
+            self.diagnostics["inv_lat"] = inv_lat
+            self.diagnostics["base_lon"] = base_lon
+            self.diagnostics["base_lat"] = base_lat
+            self.diagnostics["inv_lon_diff"] = inv_lon_diff
+            self.diagnostics["inv_lat_diff"] = inv_lat_diff
+            self.diagnostics["pixel_lon"] = pixel_lon
+            self.diagnostics["pixel_lat"] = pixel_lat
+            self.diagnostics["lon_diff"] = lon_diff
+            self.diagnostics["lat_diff"] = lat_diff
+            self.diagnostics["source_area"] = source_area
+            self.diagnostics["count"] = xr.DataArray(
+                br.get_count(), dims=("y", "x"), attrs={"area": self.base_area})
+        return (inv_lon, inv_lat)
+
+
+class ParallaxCorrectionModifier(ModifierBase):
+    """Modifier for parallax correction.
+
+    Apply parallax correction as a modifier.  Uses the
+    :class:`ParallaxCorrection` class, which in turn uses the
+    :func:`forward_parallax` function.  See the documentation there for
+    details on the behaviour.
+
+    To use this, add in your ``etc/modifiers/visir.yaml`` something like::
+
+        sensor_name: visir
+
+        modifiers:
+          parallax_corrected:
+            modifier: !!python/name:satpy.modifiers.parallax.ParallaxCorrectionModifier
+            prerequisites:
+              - "ctth_alti"
+            resampler_args:
+              radius_of_influence: 50000
+
+        composites:
+
+          parallax_corrected_VIS006:
+            compositor: !!python/name:satpy.composites.SingleBandCompositor
+            prerequisites:
+              - name: VIS006
+                modifiers: [parallax_corrected]
+
+    Here, ``ctth_alti`` is CTH provided by the ``nwcsaf-geo`` reader, so to use it
+    one would have to pass both on scene creation::
+
+        sc = Scene({"seviri_l1b_hrit": files_l1b, "nwcsaf-geo": files_l2})
+        sc.load(["parallax_corrected_VIS006"])
+
+    Alternately, you can use the lower-level API directly with the
+    :class:`ParallaxCorrection` class, which may be more efficient if multiple
+    datasets need to be corrected.  RGB Composites cannot be modified in this way
+    (i.e. you can't replace "VIS006" by "natural_color").  To get a parallax
+    corrected RGB composite, create a new composite where each input has the
+    modifier applied.
+    """
+
+    def __call__(self, projectables, optional_datasets=None, **info):
+        """Apply parallax correction.
+
+        The argument ``projectables`` needs to contain the dataset to be
+        projected and the height to use for the correction.
+        """
+        (to_be_corrected, cth) = projectables
+        base_area = to_be_corrected.attrs["area"]
+        corrector = self._get_corrector(base_area)
+        plax_corr_area = corrector(cth)
+        res = resample_dataset(
+                to_be_corrected, plax_corr_area,
+                radius_of_influence=50000, fill_value=np.nan)
```

Comment:
Computer dynamically, I'm not sure.  It's supposed to use the resampler arguments as in the docstring example, so that it doesn't is definitely a bug!

Reply:
So this needs to be fixed? As mentioned in the other radius_of_influence comment, there should be a "decent" default produced for radius_of_influence, but it definitely isn't perfect. If there is something that we can change in the near term to make it better (or to add another option to make it work better for this use case) then we should do that.

Reply:
It is configurable in 4e977673.